services/orders_find/lambda_function.py

```diff
@@ -3,33 +3,20 @@
 import uuid
 import boto3
 import decimal
-from boto3.dynamodb.conditions import Key, Attr
-from botocore.exceptions import ClientError
 
 # get the service resource
 dynamodb = boto3.resource('dynamodb')
 # get the users table
 ORDERS_TABLE = os.environ['ORDERS_TABLE']
 
-# helper class to convert a DynamoDB item to JSON
-# for details see: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GettingStarted.Python.03.html
-class DecimalEncoder(json.JSONEncoder):
-    def default(self, o):
-        if isinstance(o, decimal.Decimal):
-            if abs(o) % 1 > 0:
-                return float(o)
-            else:
-                return int(o)
-        return super(DecimalEncoder, self).default(o)
-
 def lambda_handler(event, context):
     
     orders_table = dynamodb.Table(ORDERS_TABLE)
     order_id = event['pathParameters']['order_id']
 
     try:
         response = orders_table.get_item(Key={'id': order_id})
-        res = str(json.dumps(response, cls=DecimalEncoder))
+        res = str(json.dumps(response, default=str))
```

Comment:
Is there a reason why we perform the `str` call after `json.dumps`? As far as I know, `json.dumps` already converts to a string.

Reply:
yup, there's a reason for this, it has to do with how the logs are shown in the AWS CloudWatch Logs, if you don't explicitly convert to str over there, then instead of nicely printing the json result in one log line that can be nicely expanded, it will print every row from within the json object on a separate line, which make understanding anything really hard

Reply:
But `json.dumps` already returns a string, right?

Reply:
yeh, I think something weird goes on with the AWS CloudWatch service, you could try to take out the explicit cast from here and see what I mean exactly in the logs

Reply:
@fabianishere actually, the `json.dumps` does not convert to a string but rather to a JSON string, and this is the cause of the problem

Reply:
Hmm, I cannot seem to reproduce this problem. They both produce the same output.

Reply:
oh, I think you could be right since now we have the `default=str` key in there, so that might already convert it to str (instead of JSON string), will also look into and remove the str cast if it's not needed indeed

Reply:
@fabianishere update code, the casting to str is now removed